File: api/src/domain/entities/subdomain.ts
export interface Subdomain {
	id: string;
	name: string;
	fullDomain: string;
	userId: string;
	applicationId?: string;
	databaseId?: string;
	isActive: boolean;
	createdAt: Date;
	updatedAt: Date;
}

export type CreateSubdomainInput = Omit<Subdomain, 'id' | 'createdAt' | 'updatedAt' | 'fullDomain'>;

export type UpdateSubdomainInput = Partial<
	Omit<Subdomain, 'id' | 'userId' | 'createdAt' | 'updatedAt'>
>;

File: api/internal/domain/subdomain.go
package domain

import (
	"time"

	"github.com/google/uuid"
)

type Subdomain struct {
	ID uuid.UUID `gorm:"type:uuid;default:gen_random_uuid();primairyKey"`
	Name string
	FullDomain string
	UserID uuid.UUID `gorm:"type:uuid;not null;index"`
	User User
	applications
	IsActive bool
	CreatedAt time.Time
	UpdatedAt time.Time
}
